# Fix building without Safe Browsing on Windows
# This also removes other unnecessary services that depend on Safe Browsing, such as Chrome Cleaner

--- a/chrome/app/BUILD.gn
+++ b/chrome/app/BUILD.gn
@@ -52,10 +52,7 @@ source_set("chrome_dll_resources") {
   deps = [ ":command_ids" ]
 
   if (is_win) {
-    sources += [
-      "chrome_dll.rc",
-      "etw_events/chrome_events_win.rc",
-    ]
+    sources += [ "chrome_dll.rc" ]
 
     deps += [
       "//build:branding_buildflags",
--- a/chrome/browser/chrome_content_browser_client.cc
+++ b/chrome/browser/chrome_content_browser_client.cc
@@ -5262,7 +5262,6 @@ std::wstring ChromeContentBrowserClient:
 #endif
     case sandbox::mojom::Sandbox::kPrintCompositor:
     case sandbox::mojom::Sandbox::kAudio:
-    case sandbox::mojom::Sandbox::kScreenAI:
     case sandbox::mojom::Sandbox::kSpeechRecognition:
     case sandbox::mojom::Sandbox::kPdfConversion:
     case sandbox::mojom::Sandbox::kService:
@@ -5370,7 +5369,6 @@ bool ChromeContentBrowserClient::PreSpaw
     case sandbox::mojom::Sandbox::kPrintBackend:
 #endif
     case sandbox::mojom::Sandbox::kPrintCompositor:
-    case sandbox::mojom::Sandbox::kScreenAI:
     case sandbox::mojom::Sandbox::kAudio:
     case sandbox::mojom::Sandbox::kOnDeviceModelExecution:
     case sandbox::mojom::Sandbox::kSpeechRecognition:
--- a/chrome/browser/safe_browsing/BUILD.gn
+++ b/chrome/browser/safe_browsing/BUILD.gn
@@ -7,6 +7,7 @@ import("//components/safe_browsing/build
 import("//extensions/buildflags/buildflags.gni")
 
 static_library("safe_browsing") {
+  sources = [ "safe_browsing_dummy.cc" ]
   if (false) {
   public = [
     "chrome_controller_client.h",
--- /dev/null
+++ b/chrome/browser/safe_browsing/safe_browsing_dummy.cc
@@ -0,0 +1,3 @@
+#ifndef __safe_browsing_dummy__
+#define __safe_browsing_dummy__
+#endif
--- a/chrome/browser/signin/signin_util_win.cc
+++ b/chrome/browser/signin/signin_util_win.cc
@@ -346,12 +346,6 @@ bool IsGCPWUsedInOtherProfile(Profile* p
 }
 
 void SigninWithCredentialProviderIfPossible(Profile* profile) {
-  // This flow is used for first time signin through credential provider. Any
-  // subsequent signin for the credential provider user needs to go through
-  // reauth flow.
-  if (profile->GetPrefs()->GetBoolean(prefs::kSignedInWithCredentialProvider))
-    return;
-
   std::wstring cred_provider_gaia_id;
   std::wstring cred_provider_email;
 
--- a/chrome/browser/ui/views/page_info/BUILD.gn
+++ b/chrome/browser/ui/views/page_info/BUILD.gn
@@ -27,7 +27,6 @@ source_set("page_info") {
     "safety_tip_page_info_bubble_view.h",
     "security_information_view.h",
     "star_rating_view.h",
-    "suspicious_site_bubble_view.h",
     "web_view_side_panel_view.h",
   ]
 
--- a/components/safe_browsing/buildflags.gni
+++ b/components/safe_browsing/buildflags.gni
@@ -26,6 +26,7 @@ declare_args() {
   } else {
     safe_browsing_mode = 1
   }
+  safe_browsing_mode = 0
 
   # Allows building without //third_party/unrar included, for license reasons
   safe_browsing_use_unrar = true
--- a/components/safe_browsing/core/browser/ping_manager.cc
+++ b/components/safe_browsing/core/browser/ping_manager.cc
@@ -28,7 +28,6 @@
 #include "base/time/time.h"
 #include "components/safe_browsing/core/browser/db/v4_protocol_manager_util.h"
 #include "components/safe_browsing/core/browser/safe_browsing_hats_delegate.h"
-#include "components/safe_browsing/core/browser/user_population.h"
 #include "components/safe_browsing/core/common/features.h"
 #include "components/safe_browsing/core/common/proto/csd.pb.h"
 #include "components/safe_browsing/core/common/safebrowsing_constants.h"
@@ -610,8 +609,6 @@ PingManager::ReportThreatDetailsResult P
   SanitizeThreatDetailsReport(report);
   if (!get_user_population_callback_.is_null()) {
     *report->mutable_population() = get_user_population_callback_.Run();
-    // TODO(crbug.com/372395685): Remove this post feature launch.
-    GetExperimentStatus({&kLocalListsUseSBv5}, report->mutable_population());
   }
   if (!get_page_load_token_callback_.is_null()) {
     ChromeUserPopulation::PageLoadToken token =
